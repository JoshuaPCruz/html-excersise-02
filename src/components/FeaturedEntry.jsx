import React from 'react';
import '../styles/components/FeaturedEntry.css';
import image1 from "../images/excsercise.png";

const FeaturedEntry = () => {
  return (
    <div className="FeaturedEntry">
      <div className="FeaturedEntry-info">
        <h2>Entrada destacada</h2>
        <h1>Crea un sitio web usando HTML5</h1>
        <p>
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Dolores
          neque exercitationem quaerat magni recusandae iure vitae totam
          accusamus repudiandae. Quia dolor consequuntur voluptate, repellat
          velit nihil quod eaque reprehenderit quos?
        </p>
      </div>
      <img src={image1} alt="" />
    </div>
  );
};

export default FeaturedEntry;