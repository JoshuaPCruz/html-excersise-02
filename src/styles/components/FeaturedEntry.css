.FeaturedEntry {
  display: flex;
  padding: 15px;
  margin-bottom: 20px;
  border-radius: 10px;
  background-color: var(--color-secondary-white);
}

.FeaturedEntry h2 {
  color: var(--color-secondary-gray);
  font-weight: bold;
}

.FeaturedEntry h1 {
  margin: 5px 0;
  color: var(--color-primary-red);
  font-weight: bold;
  font-size: 1.5em;
}

.FeaturedEntry img {
  width: 600px;
  padding: 5px;
  border: 1px solid var(--color-primary-gray);
}