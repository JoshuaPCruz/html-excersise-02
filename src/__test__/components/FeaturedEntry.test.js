import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import FeaturedEntry from '../../components/FeaturedEntry';

describe('<FeaturedEntry />', () => {
  const featuredEntry = shallow(<FeaturedEntry />);
  test('Render del componente FeaturedEntry', () => {
    expect(featuredEntry.length).toEqual(1);
  });

  test('Render del tÃ­tulo', () => {
    expect(featuredEntry.find('h2').text()).toEqual('Entrada destacada');
  });
});

describe('FeaturedEntry Snapshot', () => {
  test('Comprobar la UI del componente FeaturedEntry', () => {
    const featuredEntry = create(<FeaturedEntry />);
    expect(featuredEntry.toJSON()).toMatchSnapshot();
  });
});