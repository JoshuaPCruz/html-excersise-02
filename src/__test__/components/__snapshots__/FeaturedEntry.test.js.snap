// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedEntry Snapshot Comprobar la UI del componente FeaturedEntry 1`] = `
<div
  className="FeaturedEntry"
>
  <div
    className="FeaturedEntry-info"
  >
    <h2>
      Entrada destacada
    </h2>
    <h1>
      Crea un sitio web usando HTML5
    </h1>
    <p>
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Dolores neque exercitationem quaerat magni recusandae iure vitae totam accusamus repudiandae. Quia dolor consequuntur voluptate, repellat velit nihil quod eaque reprehenderit quos?
    </p>
  </div>
  <img
    alt=""
    src={Object {}}
  />
</div>
`;
